### ISSUE TITLE
Element constructed with object selector throws error if isExisting() is called

### ISSUE THREAD

**Issue by akloeber** [webdriverio/webdriverio#5789]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 6.4.5
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** v12.18.3
 - **NPM version:** 6.14.6
 - **Browser name and version:** Chrome 85.0.4183.83
 - **Platform name and version:** macOS 10.15.6
 - **Additional wdio packages used (if applicable):** *defaults when following "Getting Started" tutorial*

**Config of WebdriverIO**
*defaults when following "Getting Started" tutorial*

**Describe the bug**
When `isExisting()` is called on an element that has been contructed with a selector object the error __"Error: selector needs to be typeof `string` or `function`"__ is thrown.

**To Reproduce**
This is a minimal reproduction testcase:
```js
describe('webdriver.io page', () => {
  it('should support isExisting on element created with object selector', () => {
    browser.url('https://webdriver.io')

    const navSiteEL = browser.execute(function (_sel) {
      return document.querySelector(_sel);
    }, '.nav-site');
    const navSite = $(navSiteEL); // create element from object selector
    console.log(navSite.$$('li')[2].$('a').getText()); // -> correctly logs "Help"
    console.log(navSite.isExisting()); // -> Error: selector needs to be typeof `string` or `function`
  });
});
```
A full setup can be found in this repo: https://github.com/akloeber/webdriverio-selector-bug (just run with `npx wdio wdio.conf.js`).

**Expected behavior**
In the testcase `isExisting()` should return `true` instead of throwing an error.

**Additional context**
It works, if the element is created with a function selector instead:
```js
// ...
    const navSite = $(function () {
      return document.querySelector('.nav-site');
    });
    console.log(navSite.$$('li')[2].$('a').getText()); // -> log "Help"
    console.log(navSite.isExisting()); // -> log `true`
// ...
```

Unfortunately we can't use this in our case because we have implemented a complex lookup strategy in the function passed to `browser.execute()` that needs an argument and `browser.$` does not support passing arguments.

We have found the following workaround but that implies another roundtrip into the browser:
```js
    const navSiteEL = browser.execute(function (_sel) {
      return document.querySelector(_sel);
    }, '.nav-site');
    const navSite = $(navSiteEL).$(function () {return this;});
    console.log(navSite.$$('li')[2].$('a').getText()); // -> correctly logs "Help"
    console.log(navSite.isExisting()); // -> log `true`
```

**Comment by christian-bromann** [comment]
Hey,
thanks for filing the issue. This seems indeed like bug. We would appreciate any help on this so please help us out! Have a look at our [contributing guidelines](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) and reach out if you have any questions. Cheers!

**Comment by joshskumar** [comment]
@christian-bromann  this issue is because execute script does not return the selector part of the object. can you please help me to understand under which package executeScript is called when the following line is executed as part of webdriverio package execute.js 
       this.executeScript(script, (0, _utils.verifyArgsAndStripIfElement)(args));

**Comment by christian-bromann** [comment]
@joshskumar the `execute` command is implemented here: https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/commands/browser/execute.js. The underlying protocol command is implemented in the browser driver so WebdriverIO just sends a rest call to it and receives whatever it receives.

**Comment by L0tso** [comment]
@joshskumar @christian-bromann This issue happens because element selector was setted  as `null` [here](https://github.com/webdriverio/webdriverio/blob/master/packages/webdriverio/src/commands/browser/%24.js#L60) and [this function](https://github.com/webdriverio/webdriverio/blob/524ac7263c6e8022d2f41875b3dd626d54d52650/packages/webdriverio/src/utils/index.js#L244) doesn't have behavior for `null` selector

**Comment by christian-bromann** [comment]
This functionality is so that you can convert an element response into a WebdriverIO element object, e.g. when fetching an active element:

```js
const activeElement = browser.getActiveElement() // returns { "element-6066-11e4-a52e-4f735466cecf": "ELEMENT-1" }
const wdioElem = $(activeElement)
wdioElem.click()
```

Same happens if you return an DOM node from the execute command. In these cases we have no information about the selector because the element was received without it. I am not sure what would be the best way to work around this. We could use the function body as selector and just re-execute the same method. If we don't even have that we could just assume someone used the `getActiveElement` method. What do you think?

**Comment by shinxi** [comment]
The simple script to reproduce the issue is
```js
const navSite = browser.execute(()=> document.querySelector(''.nav-site''));
console.log(navSite.isExisting()); // -> Error: selector needs to be typeof `string` or `function`
```

However, it doesn't seem necessary to do the isExisting assertion which will always return true in this case, I tend to think doing null check will make more sense, i.e.,
```js
const navSite = browser.execute(()=> document.querySelector(''.nav-site''));
expect(navSite).not.toBeNull();
```

**Comment by christian-bromann** [comment]
@shinxi you can't do this:

> ```js
> const navSite = browser.execute(()=> document.querySelector(''.nav-site''));
> console.log(navSite.isExisting()); // -> Error: selector needs to be typeof `string` or `function`
> ```

because `navSite` is not an WebdriverIO element. You should get `TypeError: navSite.isExisting is not a function` with the latest version.

**Comment by christian-bromann** [comment]
After re-visiting this issue it seems that this got fixed by #7348. The following example now works for me:

```js
const { remote } = require('webdriverio')

let browser

;(async function () {
    browser = await remote({
        capabilities: {
            browserName: 'chrome',
        }
    })

    await browser.url('https://webdriver.io')

    const navSiteEL = await browser.execute(function (_sel) {
        return document.querySelector(_sel);
    }, '.footer__items');
    const navSite = await browser.$(navSiteEL);
    console.log(await navSite.$$('li')[2].$('a').getText());
    console.log(await navSite.isExisting());

    await browser.deleteSession()
})().catch(async (e) => {
    console.error(e.stack)
    await browser.deleteSession()
})

```

**Comment by shinxi** [comment]
> because navSite is not an WebdriverIO element. You should get TypeError: navSite.isExisting is not a function with the latest version.

Oh, yes, I missed $ in my sample. Thanks @christian-bromann 
```js
const navSite = browser.execute(()=> document.querySelector(''.nav-site''));
console.log($(navSite).isExisting()); // -> Error: selector needs to be typeof `string` or `function`
```
What I meant is, `expect(navSite).not.toBeNull()` will make more sense than `expect($(navSite).isExisting()).toBe(true)`.
